src/index.c

```diff
@@ -1139,6 +1159,13 @@ static int index_insert(
 		 * check for dups, this is actually cheaper in the long run.)
 		 */
 		error = git_vector_insert_sorted(&index->entries, entry, index_no_dups);
+
+		if (error == 0) {
+			if (index->ignore_case)
+				git_idxmap_icase_insert((khash_t(idxicase) *) index->entries_map, entry, entry, error);			else
+				git_idxmap_insert(index->entries_map, entry, entry, error);
```

Comment:
Yeah after having read the rest of the PR I understand what this was intended to do now.  :P

I wonder if we might be able to have a macro or something to avoid all the `index->ignore_case switches`.  (But then we have two problems.)